build-common/NHibernate.props

```diff
@@ -2,13 +2,14 @@
   <Import Condition="Exists('NHibernate.dev.props')" Project="NHibernate.dev.props"/>
 
   <PropertyGroup>
-    <VersionMajor Condition="'$(VersionMajor)' == ''">5</VersionMajor>
-    <VersionMinor Condition="'$(VersionMinor)' == ''">2</VersionMinor>
-    <VersionPatch Condition="'$(VersionPatch)' == ''">7</VersionPatch>
-    <VersionSuffix Condition="'$(VersionSuffix)' == ''"></VersionSuffix>
+    <NhVersion     Condition="'$(NhVersion)' == ''"    >5.3</NhVersion>
+    <VersionPatch  Condition="'$(VersionPatch)'  == ''">0</VersionPatch>
+    <!-- Clear VersionSuffix for making release and set it to dev for making development builds -->
+    <VersionSuffix Condition="'$(VersionSuffix)' == ''">dev</VersionSuffix>
 
-    <VersionPrefix>$(VersionMajor).$(VersionMinor).$(VersionPatch)</VersionPrefix>
-    <AssemblyVersion>$(VersionMajor).$(VersionMinor).0.0</AssemblyVersion>
+    <VersionPrefix Condition="'$(VersionPrefix)' == ''">$(NhVersion).$(VersionPatch)</VersionPrefix>
+    <VersionSuffix Condition="'$(VersionSuffix)' != '' AND '$(BuildNumber)' != ''">$(VersionSuffix).$(BuildNumber)</VersionSuffix>
+    <AssemblyVersion>$(NhVersion).0.0</AssemblyVersion>
     <FileVersion>$(VersionPrefix).0</FileVersion>
```

Comment:
We are back to previous `FileVersion` handling, so ok on principle.

Nonetheless this PR is also for preparing the ability to push pre-release packages. So I wonder, shouldn't we also take care of releasing a unique `FileVersion` for these pre-releases? Otherwise, someone distributing a pre-release through an installer may end up not being able to upgrade it until we change at least the patch version.

So maybe change `.0` by `.*` if that works here, or put in fourth number the TC build number. Or maybe just stay like this, not supporting packing pre-releases. What do you think?

Reply:
I've checked how it's done by microsoft on `Microsoft.Data.SqlClient` example:
https://www.nuget.org/packages/Microsoft.Data.SqlClient/2.0.0-preview3.20122.2

Yeah build number is part of file version - so yes let's add it too. Though they also don't bother with keeping assembly version stable for minor releases...

Reply:
Keeping assembly version stable is important with strong naming, which NHibernate still issues, while it seems many do no more bother with strong naming.
But ceasing to be strongly named is a breaking change, so we stay strongly named.